package.json

```
@@ -6,6 +6,34 @@
     "name": "Arjun Vegda",
     "url": "https://arjunvegda.com"
   },
+  "sideEffects": false,
+  "source": "./src/index.ts",
+  "exports": {
+    ".": {
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.esm.js"
+      },
+      "require": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.cjs.js"
+      }
+    },
+    "./utils": {
+      "import": {
+        "types": "./dist/utils.d.ts",
+        "default": "./dist/utils.esm.js"
+      },
+      "require": {
+        "types": "./dist/utils.d.ts",
+        "default": "./dist/utils.cjs.js"
+      }
+    },
+    "./package.json": "./package.json"
+  },
```

my understanding is mjs/mts extensions are required.

here's my suggestion. probably we should have "default".
```suggestion
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.esm.mjs"
      },
      "module": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.esm.js"
      },
      "default": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.cjs.js"
      }
    },
    "./*": {
      "import": {
        "types": "./dist/*.d.mts",
        "default": "./dist/*.esm.mjs"
      },
      "module": {
        "types": "./dist/*.d.ts",
        "default": "./dist/*.esm.js"
      },
      "default": {
        "types": "./dist/*.d.ts",
        "default": "./dist/*.cjs.js"
      }
    }
  },
```